### DISCUSSION TITLE
GSoC 2025: expression of interest

### DISCUSSION THREAD

**Discussion by thibaudcolas** [wagtail/gsoc#123] [Announcements]
👋 we’ve not decided whether we’ll join GSoC in 2025. If you want to participate with us, please register your interest with [our form: wagtail.org/gsoc-apply](https://wagtail.org/gsoc-apply/)

## Stay up to date

To know once we participate, we’ll announce on our [This week in Wagtail newsletter](https://wagtail.org/newsletter/), [Mastodon account](https://fosstodon.org/@wagtail), [LinkedIn page](https://www.linkedin.com/company/wagtail-cms/), [Bluesky account](https://bsky.app/profile/wagtail.org), and at any [Wagtail events](https://wagtail.org/events/) we have in early 2025, and on the Wagtail Slack. And here :)

## Timeline

At this stage we don’t even know whether GSoC as a program will be running in 2025, but here’s what to expect based on the 2024 timeline. **TL;DR:** sit tight until March 2025.

- In early January 2025, we’ll know whether GSoC is happening at all or not.
- In late January, we’ll be able to share whether Wagtail is _applying_ to participate, or not.
- In late February, we’ll be able to share whether we’ve been accepted to participate.
- In mid-March 2025 – we’ll have a list of provisional project ideas and start the contributor application period.

## 2025 projects

At this stage all I can say for sure is we’ll want to have two to four GSoC contributors, with four to ten mentors, representing a good diversity of genders, backgrounds, countries. We’ll be looking for women / non-binary contributors in particular.

As far as hot topics for projects, we forecast: accessibility, sustainability, AI, performance, deployments, code health, security.

**Comment by pooranjoyb** [comment]
would be glad to be a part of wagtail. however no updates of gsoc 2025 yet.

**Comment by vinayJ74** [comment]
@thibaudcolas  but why do you say that if gsoc as a program will run in 2025

**Reply by thibaudcolas** [reply]
Google hasn’t announced it yet, as far as I know they decide on a yearly basis, we can’t really plan for anything specific until they announce it.

**Comment by ayaan-qadri** [comment]
@thibaudcolas, Is there anything where i can start contributing ?

**Reply by thibaudcolas** [reply]
We always encourage new contributors to get involved, currently you can refer to our existing docs about starting contributions

**Comment by Gokulan-SK** [comment]
Thanks for the update! I’m definitely interested in contributing and will keep an eye out for announcements ☺️

**Comment by thibaudcolas** [comment]
We discussed our GSoC participation in an info session today, and reiterated we’d be looking to participate next year, with 2-3 projects :)

![Community hour attendees on 2024-10-25 Google Meet screenshot with 6 attendees](https://github.com/user-attachments/assets/484ad0e3-4757-469d-a55d-a23537a3e3e3)

## Project ideas

We discussed the __security__ ideas specifically, where I mentioned some of the possible topics within this theme:

- [Content security Policy (CSP) support](https://github.com/wagtail/wagtail/issues/1288)
- Making more of automated code scanning features from GitHub
- Setting up fuzz testing, possibly within [oss-fuzz](https://github.com/google/oss-fuzz)
- For someone with the right expertise - doing a more manual independent security audit of Wagtail, and then addressing some of the issues.

@laymonage also raised [Stimulus adoption](https://github.com/orgs/wagtail/projects/14/views/1?query=sort%3Aupdated-desc+is%3Aopen) as something we could have as a theme (ties in CSP)

@PaarthAgarwal is interested to mentor on projects relating to sustainability. There’s a good chance we’ll have a project in this field, likely about low-carbon emissions websites (in particular carbon-aware website concepts).

## Contributors getting started

We said having upfront knowledge of Django and Wagtail isn’t necessarily an upfront requirement – but it certainly helps to know that as a baseline. We recommend people consider going through the [Wagtail tutorial](https://docs.wagtail.org/en/stable/getting_started/tutorial.html) or [Django Girls tutorial](https://tutorial.djangogirls.org/en/) (or both)!

**Reply by ayaan-qadri** [reply]
Thanks for the update @thibaudcolas , looking forward to be part of next year and excited as well, Let's see !

**Reply by kelue** [reply]
Thanks for keeping us up to date. Patiently waiting on any news as I am very interested in contributing to wagtail for GSOC 2025.

**Comment by dharmendra-byte** [comment]
Thank you, Sir, for the update, @thibaudcolas. I'm looking forward to being a part of next year’s events and am truly excited about the opportunities ahead. Let’s see what unfolds!

**Comment by Valley-15** [comment]
Hello everyone!!

I am Adnaan, a third-year CSE student from VNRVJIET, Hyderabad. I am highly interested in contributing to open source and growing alongside like-minded people. I have skills in the MERN stack, along with familiarity in Python, HTML, CSS, and JavaScript. I was selected for the Girl Script Summer of Code and have successfully merged 4+ PRs during Hacktoberfest. I’m eager to explore, learn, and make impactful contributions to open source projects!  

If you'd like to connect and collaborate, feel free to reach out to me at adnaan.2735@gmail.com! Looking forward to growing together!

**Comment by suryanshgargbpgc** [comment]
Hi @thibaudcolas, I am new to opensource, I have a background in Python and Django, can you share documentations, which can help me understand codebase of Wagtail.

**Reply by lb-** [reply]
@suryanshgargbpgc everything you need is in our docs. 

You can start here https://docs.wagtail.org/en/stable/contributing/first_contribution_guide.html

**Reply by suryanshgargbpgc** [reply]
Thanks a lot !!

**Comment by snehajain1011** [comment]
I am interested in making meaningful contributions and look forward to receiving updates.

**Comment by adot-7** [comment]
Hi @thibaudcolas and other maintainers,
I’m new to open source but have some experience with Django, and I’m really excited to learn and hopefully contribute to this project for GSoC'25. Outside of coding, I enjoy playing football and brushing up on my French (though I’m not sure how that connects to Wagtail).

**Comment by Bhanu-partap-13** [comment]
Hi @thibaudcolas , I am new to opensource .However, I have a background in Python and Javascript. I would be glad to contribute in wagtail.

**Comment by keshxvdayal** [comment]
### Hello @thibaudcolas, 
I'm Keshav Dayal, a third-year BCA student and a Full Stack Developer with a passion for open-source contributions. My tech stack includes Python, Django, React, Next.js, Tailwind CSS, and TypeScript, with additional expertise in Flask, SQLite, and Sanity CMS. I have substantial experience working on scalable projects, such as building an e-commerce website with 400+ products and leading the development of Organizations. My internships have enhanced my skills in optimizing performance and user engagement. I'm excited to contribute to Wagtail by leveraging my technical expertise and commitment to impactful development.

Patiently waiting on any news as I am very interested in contributing to wagtail for GSOC 2025.

-> Please Do Let Me know if i can start contribution in any feature, functionality or anything which helps you organization enchancing the workflow.

**Comment by anurag6569201** [comment]
hii @thibaudcolas  very excited to inform you that going to start contribute to the  Wagtail with having good experience in django azure and ml going to use those knowledge and rectify the issues

**Comment by surya-ansh** [comment]
Hi @thibaudcolas,
I’m Suryansh, new to open-source, but I have a background in Python and JavaScript. I’m really interested in contributing to Wagtail and would love some guidance on where to start or how I can be of help.

**Comment by nilaygit-10721** [comment]
Hello @thibaudcolas, I am starting my contribution to Wagtail.

**Comment by GovindHede** [comment]
Hello @thibaudcolas ,
I’m Govind Hede, a Computer Science student from Pune University and an aspiring GSoC contributor. I have experience in Python, Django, Flask, and Docker, with a strong focus on backend development. I’ve worked on building scalable projects that integrate RESTful APIs and WebSockets, and I’m passionate about creating clean, efficient, and maintainable code.

I’m excited about the opportunity to contribute to Django and be a part of this incredible community. Django has been a significant part of my learning journey, and I’m eager to give back by collaborating on impactful projects, learning from mentors, and growing as a developer.

Looking forward to contributing and learning alongside you all! 😊

**Comment by Mohamed-Rabiaa** [comment]
Hi, I am Mohamed Rabiaa, a backend software engineer from Egypt with experience building Python and Django projects. I am very motivated to start contributing to Wagtail and have an impact on such a reputable CMS platform.

**Comment by thibaudcolas** [comment]
👋 just a quick update that we’ve not heard from Google yet about whether there will be a GSoC 2025. I expect it will be announced [on the Google Open Source blog](https://opensource.googleblog.com/search/label/gsoc). Even once that’s announced, we won’t know if Wagtail takes part until _late February_ (25th or so).

For now – best thing for people to do is register their interest here: [wagtail.org/gsoc-apply](https://wagtail.org/gsoc-apply/). Looking forward to this!

**Comment by rahul-vyas-dev** [comment]
Hi @thibaudcolas, Whether or not you participate in GSoC 2025, I just want to say how great it feels to be part of this community and learn from people like you. If you do take part, it would be amazing to contribute alongside you. But even if you don’t, I’ll keep contributing, learning from mentors, and connecting with others who have so much knowledge and experience to share.

**Reply by thibaudcolas** [reply]
🤘 even if we don’t take part there’ll be plenty other cool orgs in the program I’m sure

**Comment by rahul-vyas-dev** [comment]
That's great if you take participant in gsoc then I will be a lucky to
becoming a part of you.
But if don't take participant in gsoc then also I will be a Lucky to
becoming a part of your team, having a great mentors like you and a kind
heart.

I will also contribute continuosly to your organization.

**Comment by Menna-Ahmed7** [comment]
I’d like to know if you’re participating in GSOC 2025. If so, I’d be thrilled to collaborate with you!

**Reply by GovindHede** [reply]
let's do it

**Comment by BiradarScripts** [comment]
let's do it


________________________________

**Comment by gschandrasekhar** [comment]
Hi
I’m Chandra Sekhar, a CS student passionate about open-source development. I recently came across wagtail.

I have experience in MERN Stack.I'm eager to contribute, learn, and collaborate with the community.
Could you guide me on how to get started with contributions? I’d love to understand more about ongoing projects and any beginner-friendly issues. Let's connect on [Linkedin ](https://www.linkedin.com/in/gschandrasekhar/)and [Twitter](https://x.com/gschandrasekhr)

**Comment by Abhishek-kumar0503** [comment]
Hello @thibaudcolas,
My self Abhishek Kumar  and final year student of Delhi University. I have a strong background in Python, Django, and AI-driven solutions, making me a great fit for this project. My experience includes developing machine learning models for text classification and working on AI-powered accessibility tools, such as a BERT-based model that detects AI-generated text with 95.06% accuracy.

I have also built a Django-based platform for translating YouTube captions, improving accessibility for non-English speakers. My skills in frontend and backend development, along with API integration, allow me to contribute effectively to enhancing Wagtail’s accessibility features.

I am excited about this project as it aligns with my passion for AI, accessibility, and open-source contributions. My experience in modifying models, writing test cases, and integrating AI solutions will help make Wagtail more inclusive and user-friendly for visually impaired users. 

Is there anything where i can start contributing ? ....

**Comment by raghulchandramouli** [comment]
Would be like to part of WagTail and in GSOC

**Comment by Anil951** [comment]
Hello @thibaudcolas  ,
I’m Anil, a final-year student passionate about AI, data science, and web development.

I’ve worked on projects using these skills. My expertise in frontend and backend development, along with API integrations, makes me a strong fit for contributing to Wagtail’s accessibility and AI-driven features.

I’m excited about this project and eager to contribute. Where can I start?
GitHub: https://github.com/Anil951

**Comment by peter-abhinav** [comment]
Hey, I'm new to open source and interested in starting my open-source journey. What should I do, and how do I get started in wagTail?

**Comment by Thanshir** [comment]
Hello @thibaudcolas ,

I'm Thanshir Mohammed, a backend developer with a strong foundation in Python, Django, and DevOps. I recently came across the Wagtail CSP Compatibility project and found it really interesting.

I have experience working with Django-based web applications, containerization with Docker, AWS deployments, CSP and frontend security, I am eager to learn and contribute towards improving Wagtail’s security.

I have some ideas on how to approach CSP implementation and would love to discuss how I can start contributing. Could you guide me on the best way to get involved? Also, are there any beginner-friendly issues I can work on to get familiar with the Wagtail codebase?

Looking forward to collaborating with the Wagtail community!

Best,
Thanshir Mohammed

**Comment by BANVEN-subzero** [comment]
Hello @thibaudcolas 
 my name is Banven Eric a software engineering undergraduate at the college of technology in the university of Buea (Cameroon) i am new to open source but I have knowledge in python , JavaScript, bootstrap. I will love to be part of the wagtail project and equally learn new skills and have more experience under your good guidance

**Comment by Jonathon-AR** [comment]
Hello folks,
This is Aarnav. I'm a 2024 CS graduate, I've professional work ex interning as a Full Stack Developer in React, Redux, Django and Springboot.
I look forward to joining the cause and contributing to the project.

**Comment by SamarAhwaz** [comment]
Hello @thibaudcolas ,

I'm Samar, a backend developer with a strong foundation in Python, Django, and Javascript I recently came across the Wagtail CSP Compatibility project and found it really interesting.

I have experience working with Django-based web applications, containerization,
I am eager to learn and contribute towards improving Wagtail’s security.

I have some ideas on how to approach CSP implementation and would love to discuss how I can start contributing. Could you guide me on the best way to get involved? Also, are there any beginner-friendly issues I can work on to get familiar with the Wagtail codebase?

Looking forward to collaborating with the Wagtail community!

Best,
Samar

**Comment by Ramosh99** [comment]
Hello @thibaudcolas ,

I’m Yasiru Ramosh, an IT undergraduate from the University of Moratuwa and an aspiring GSoC contributor. I have experience in Docker, Kubernetes, FastAPI, Django, Java, JavaScript-based frameworks (React, Next.js), and RASA, along with hands-on work in Retrieval-Augmented Generation (RAG).

I’m really excited about contributing to Wagtail and being part of this vibrant open-source community. Could you guide me on how I can get started and contribute? Also, what are some available tasks or projects that align with my skill set?

**Comment by Boltworks-lab** [comment]
I am definitely interested, been looking for a python based open source project to contribute to, since I did an open source course last semester. This is absolutely perfect for me.

**Comment by 4444Harsh** [comment]
Hello @thibaudcolas, Whether or not you join GSoC 2025, I just wanted to say thanks for being part of this wonderful community. Being able to learn from seasoned contributors like yourself has been inspiring. If you do join, I'd love the chance to collaborate with you. But even if not, I'll keep learning, contributing, and interacting with the wonderful mentors and developers here.

**Comment by utkarshgupta2804** [comment]
I’m new to open source but eager to contribute to your Content Security Policy compatibility project, matching my cybersecurity passion and web development skills.

**Comment by Haiderkhan64** [comment]
Hi @thibaudcolas and everyone! I’m Haider Khan, new to Wagtail and open source. I’m really excited about the Grid-Aware Websites project for GSoC 2025—making websites sustainable sounds awesome! I haven’t applied yet but plan to this week. Just started the Wagtail tutorial to get a feel for it—any tips on how to prep for the project or join in? Maybe something about Cloudflare Workers? Can’t wait to learn and contribute with you all!
